File: 1_PythonServer/app/domain/llm/generation_params.py
from dataclasses import dataclass
from enum import StrEnum


class LlmOutputFormat(StrEnum):
    TEXT = "text"
    JSON_OBJECT = "json_object"


class LlmReasoningMode(StrEnum):
    DEFAULT = "default"
    AUTO = "auto"
    ENABLED = "enabled"
    OFF = "off"
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    MAX = "max"


@dataclass(frozen=True, slots=True)
class LlmReasoningOptions:
    mode: LlmReasoningMode = LlmReasoningMode.DEFAULT
    budget_tokens: int | None = None


@dataclass(frozen=True, slots=True)
class LlmGenerationParams:
    temperature: float | None = None
    top_p: float | None = None
    presence_penalty: float | None = None
    frequency_penalty: float | None = None
    max_output_tokens: int | None = None
    reasoning: LlmReasoningOptions | None = None


@dataclass(frozen=True, slots=True)
class LlmOutputOptions:
    format: LlmOutputFormat = LlmOutputFormat.TEXT

File: 2_ReactWeb/src/entities/llm-runtime/model/generationParams.ts
export type DsLlmReasoningMode =
  | "default"
  | "auto"
  | "enabled"
  | "off"
  | "low"
  | "medium"
  | "high"
  | "max";

export type DsLlmOutputFormat = "text" | "json_object";

export type DsLlmReasoningOptions = {
  budgetTokens?: number;
  mode: DsLlmReasoningMode;
};

export type DsLlmGenerationParams = {
  frequencyPenalty?: number;
  maxOutputTokens?: number;
  presencePenalty?: number;
  reasoning?: DsLlmReasoningOptions;
  temperature?: number;
  topP?: number;
};

export type DsLlmOutputOptions = {
  format: DsLlmOutputFormat;
};
